clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1223,6 +1233,26 @@ public void maybeAutoCommitOffsetsAsync(long now) {
         }
     }
 
+    private boolean waitForPendingAsyncCommits(Timer timer) {
+        if (inFlightAsyncCommits.get() == 0) {
+            return true;
+        }
+
+        do {
+            ensureCoordinatorReady(timer);
```

Comment:
commitOffsetSync is already attempting to connect to the coordinator, so it might make more sense to remove this, and move the` if (offset.isEmpty()) {...} `  below `coordinatorUnknownAndUnreadySync`.

Reply:
scratch this off, see the comment above.

Reply:
See this comment: https://github.com/apache/kafka/pull/13678#discussion_r1190730213